src/librustc/ty/mod.rs

```diff
@@ -3349,13 +3349,29 @@ fn issue33140_self_ty(tcx: TyCtxt<'_>, def_id: DefId) -> Option<Ty<'_>> {
     }
 }
 
+/// Check if a function is async.
+fn asyncness(tcx: TyCtxt<'_>, def_id: DefId) -> hir::IsAsync {
+    if let Some(hir_id) = tcx.hir().as_local_hir_id(def_id) {
+        let node = tcx.hir().get(hir_id);
+        if let Some(fn_like) = hir::map::blocks::FnLikeNode::from_node(node) {
```

Comment:
Similarly here I would probably just `unwrap` the result, as this query really ought to only be invoked for fn-like def-ids:

```rust
let fn_like = hir::map::blocks::FnLikeNode::from_node(node).unwrap_or_else(|| {
    bug!("asyncness: expected fn-like node but got `{:?}`", def_id);
});
```